src/viewer.js

```diff
@@ -2477,31 +2477,48 @@ function onBlur(){
 }
 
 function onCanvasKeyDown( event ) {
-    if ( !event.preventDefaultAction && !event.ctrl && !event.alt && !event.meta ) {
+    var canvasKeyDownEventArgs = {
+      originalEvent: event.originalEvent,
+      preventDefaultAction: event.preventDefaultAction,
+      preventVerticalPan: event.preventVerticalPan,
+      preventHorizontalPan: event.preventHorizontalPan
+    };
+
+    this.raiseEvent( 'canvas-key-down', canvasKeyDownEventArgs);
```

Comment:
I like that we're using just one event name instead of differentiating between key down and key press (which seems like it would be unnecessary in our context). I'm a little concerned that calling it `canvas-key-down` implies that it's specifically a key down event. Maybe we should call it just `canvas-key`?

Reply:
Yes I used the same name so it's more "transparent" for the user, that does not need to know about the difference in the backend.

Sure, I agree about changing the event name. ;) I